kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveStore.java

```diff
@@ -100,176 +73,110 @@ public void init(final ProcessorContext context, final StateStore root) {
-    // needs to create and lock state directories. since
-    // the Responsive Client doesn't require flushing state
-    // to disk, we return false even though the store is
-    // persistent in a remote store
-    return false;
+  public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
+    delegate.putAll(entries);
   }
 
   @Override
-  public void put(final Bytes key, final byte[] value) {
-    buffer.put(key, value);
-    StoreQueryUtils.updatePosition(position, context);
+  public byte[] delete(final Bytes key) {
+    return delegate.delete(key);
   }
 
   @Override
-  public byte[] putIfAbsent(final Bytes key, final byte[] value) {
-    // since there's only a single writer, we don't need to worry
-    // about the concurrency aspects here (e.g. it's not possible
-    // that between the get(key) and put(key, value) another write
-    // comes in unless this client is fenced, in which case this
-    // batch will not be committed to remote storage)
-    final byte[] old = get(key);
-    if (old == null) {
-      put(key, value);
-    }
-    return old;
+  public String name() {
+    return name.kafkaName();
   }
 
   @Override
-  public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
-    entries.forEach(kv -> put(kv.key, kv.value));
-    StoreQueryUtils.updatePosition(position, context);
+  public void flush() {
+    delegate.flush();
   }
 
   @Override
-  public byte[] delete(final Bytes key) {
-    // single writer prevents races (see putIfAbsent)
-    final byte[] old = get(key);
-    buffer.tombstone(key);
-    return old;
+  public void close() {
+    delegate.close();
   }
 
   @Override
-  public byte[] get(final Bytes key) {
-    // try the buffer first, it acts as a local cache
-    // but this is also necessary for correctness as
-    // it is possible that the data is either uncommitted
-    // or not yet pushed to the remote store
-    final Result<Bytes> result = buffer.get(key);
-    if (result != null) {
-      return result.isTombstone ? null : result.value;
-    }
-
-    return client.get(name.cassandraName(), partition, key);
+  public boolean persistent() {
+    // NOTE: this is called before the store is initialized
+    // because streams uses whether the store is persistent
+    // while building the topology
+    return false;
   }
 
   @Override
-  public KeyValueIterator<Bytes, byte[]> range(final Bytes from, final Bytes to) {
-    return new LocalRemoteKvIterator<>(
-        buffer.range(from, to),
-        client.range(name.cassandraName(), partition, from, to),
-        PLUGIN
-    );
+  public boolean isOpen() {
+    return delegate != null && delegate.isOpen();
   }
 
   @Override
-  public KeyValueIterator<Bytes, byte[]> all() {
-    return new LocalRemoteKvIterator<>(
-        buffer.all(),
-        client.all(name.cassandraName(), partition),
-        PLUGIN
-    );
+  @Evolving
+  public <R> QueryResult<R> query(final Query<R> query, final PositionBound positionBound,
```

Comment:
nit: formatting

(maybe this is another thing I just assumed, but we're standardizing on separate lines per parameter for long lines, right? If not, that would be my vote)

Reply:
👍 yes we are, these are just auto-generated and sometimes IntelliJ doesn't respect my style guide... generally though I feel that if checkstyle doesn't have a check for it we shouldn't enforce it on PRs - so either we find a way to enforce it with a custom checkstyle or we're just happy go lucky the way we are